src/solve.jl

```diff
@@ -5,7 +5,7 @@ function DiffEqBase.__solve(prob::DiffEqBase.AbstractRODEProblem,
                             kwargs...) where recompile_flag
   integrator = DiffEqBase.__init(prob,alg,timeseries,ts,recompile;kwargs...)
   solve!(integrator)
-  integrator.sol
+  deepcopy(integrator.sol)
```

Comment:
deepcopying the integrator is never the solution 😱

Reply:
Ok. I understand that.

Do you have any solution in mind? Or are you ok as it is?

I just wanna know if I keep trying this or not. Part of me want to let go, but the other part keeps trying 😂 

Is deepcopying just the noise part of the integrator ok? What if I `W = deepcopy(prob.noise)` at the beginning of solve, and then, before exiting the solver, at line 8 above, I do
```julia
solve!(integrator)
if typeof(prob) <: DiffEqBase.AbstractRODEProblem
   noise_deepcopy!(prob.noise, integrator.sol.W)
end
integrator.sol
```
where `noise_deepcopy!` would be defined at `DiffEqNoiseProcess.jl` as
```julia
function noise_deepcopy!(W::T, Z::T) where {T <: AbstractNoiseProcess}
    for x in fieldnames(T)
        setfield!(W, x, deepcopy(getfield(Z, x)))
    end
end
```

Although `prob` is immutable, all noises are mutable and this works fine. All tests are passing (except those that always hang on my machine...)

Just let me know before I trigger more gh actions. 😬